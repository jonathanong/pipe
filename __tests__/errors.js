
// const { PassThrough } = require('stream')
const assert = require('assert')

const { streamToString } = require('../utils')
const render = require('../lib')

test('a number should throw', async () => {
  try {
    await streamToString(render`1${2}3`)
  } catch (err) {
    assert(/value passed/i.test(err.message))
    return
  }
  throw new Error('boom')
})

test('a sync function error should be caught and thrown in the stream', async () => {
  const fn = () => { throw new Error('boom') }
  const errors = []
  const stream = render`1${fn}2`
  stream.on('error', err => errors.push(err))
  const result = await streamToString(stream, false)
  assert.strictEqual(result, '12')
  expect(result).toMatchSnapshot()
  assert.strictEqual(errors.length, 1)
  assert.strictEqual(errors[0].message, 'boom')
})
