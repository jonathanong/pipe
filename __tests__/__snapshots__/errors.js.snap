// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`a sync function error should be caught and thrown in the stream 1`] = `"12"`;
